pkg/controlplane/client.go

```diff
@@ -113,11 +114,19 @@ func (c *client) IsActive() bool {
 
 // setActive the peer status (active or not).
 func (c *client) setActive(active bool) {
-	c.lock.Lock()
-	defer c.lock.Unlock()
-	c.active = active
-	if active || c.lastSeen.IsZero() {
-		c.lastSeen = time.Now()
+	var activePrevState bool
+	{
+		c.lock.Lock()
+		defer c.lock.Unlock()
```

Comment:
I don't think that a block helps to make this defer run when you want it.
i.e. https://stackoverflow.com/questions/49456943/why-is-gos-defer-scoped-to-function-not-lexical-enclosure
Let's revert this block, and instead call unlock directly at the end of it.